freediscovery/interpretation.py

```diff
@@ -0,0 +1,170 @@
+    Deduplicate overlapping positions by keeping the longest possible keywords.
+
+    Example 1:
+        document text contains string "email me at eric_bass@some-email.com"
+        keywords vocabulary has "eric", "bass", "eric_bass"
+        Thus keywords positions will have three tuples [(12, 16), (17, 21), (12, 21)].
+        As we cannot overlap color spans, we have to deduplicate.
+        In this examples we could either keep "eric" and "bass" or to keep "eric_bass" only and delete two others.
+        We make a choice to keep the longest possible substring, deleting every other keyword overlapping with it.
+        In this example the final output after deduplication will be [(12, 21].
+
+    Example 2:
+        document text contains string "administrative procedures"
+        keywords vocabulary has "administrative", "mini", "procedure", "pro"
+        Thus keywords positions are [(0, 14), (2, 6), (15, 24), (15, 18)].
+        Keeping longest possible keywords will eliminate "mini" and "pro" keywords,
+        which were just accidentally matched and would represent a noise if kept.
+        So, after deduplication only two tuples are left: [(0, 14), (15, 24)]
+    """
+    overlap = True
+    while overlap:
+        overlap = False
+        for idx in range(len(positions)-1):
+            current_start, current_end = positions[idx]
+            next_start, next_end = positions[idx+1]
+            if current_end > next_start:
+                overlap = True
+                # In case of overlap, keep the longer substring,
+                # and delete the indices of a shorter one from the list
+                # Then break from the for-loop and restart analyzing the list
+                if (current_end - current_start) > (next_end - next_start):
+                    del positions[idx+1]
+                else:
+                    del positions[idx]
+                break
+    return sorted(positions)
+
+
+def _wrap_in_colored_span(word, score):
+    rgba = _score_to_rgb(score, COLORMAP)
+    return '<span style="background-color: rgba({}, {}, {}, {});">{word}</span>'.format(*rgba, word=word)
+
+
+def _score_to_rgb(score, colormap):
+    mapped_rgba = colormap(score)
+    r, g, b = [int(255*x) for x in list(mapped_rgba)[:3]]
+    return r, g, b, mapped_rgba[-1]
+
+
+"""
+import numpy.random as rnd
+from sklearn.feature_extraction.text import CountVectorizer
+
+def _create_weights(filename):
+    vect = CountVectorizer(input="filename")
+    _ = vect.fit([filename])
+    features_weight = {word: rnd.random() for word in vect.vocabulary_}
+    return features_weight
+
+if __name__ == "__main__":
+    fname = 'data/ds_001/raw/0.7.6.28635.txt'
+    words_weights = _create_weights(fname)
+    print(list(words_weights.items())[:3])
+
+    with open(fname, encoding='utf-8') as in_file:
+        document_lines = [line.strip() for line in in_file.readlines()]
+
+    document_html = explain_logistic_regression(words_weights, document_lines)

```

Comment:
Actually could you add this line as another unit test please, it would require a `colormap` argument (which would need matplotlib), so to avoid importing matplotlib in unit tests we could do something along the lines of,
```py
def test_explain_categorization():

    from unittest import SkipTest
    try:
        from unittest.mock import patch, MagicMock
    except ImportError:
        raise SkipTest
    mm = MagicMock()
    #  [...]
    document_html = explain_categorization(words_weights, document_lines, colormap=mm)
```